src/main/java/io/takari/maven/builder/smart/ReactorBuildQueue.java

```diff
@@ -43,8 +43,8 @@ public ReactorBuildQueue(Collection<MavenProject> projects,
       }
     }
 
-    this.rootProjects = ImmutableSet.copyOf(rootProjects);
-    this.projects = ImmutableSet.copyOf(projects);
+    this.rootProjects = unmodifiableSet(rootProjects);
```

Comment:
not copy of?

Reply:
Intentional, because copyOf doesn't create a real copy, it is closer to a view and I wanted to keep behaviour.